### ISSUE TITLE
next dev --turbo only correctly compiles one route group having multiple root layouts in each of them

### ISSUE THREAD

**Issue by razenization** [vercel/next.js#57727]
### Link to the code that reproduces this issue

https://github.com/razenization/turbo-bug-repro

### To Reproduce

1. Start the application using pnpm dev (next dev --turbo)
2. Try to visit test1, test2 routes
3. Notice how one of them (for me in particular it was test2) renders successfully, while the other one fails to compile and throws weird error in the console saying that some DOM operations failed
4. Try removing the --turbo flag and see how things are back to normal

### Current vs. Expected behavior

I expected to be able to visit both test1 and test2 routes successfully, while having two separate root layouts in two route groups. This setup works if application is started without using the --turbo flag and I am able to open both routes.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:28 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T8110
Binaries:
  Node: 20.7.0
  npm: 10.1.0
  Yarn: N/A
  pnpm: 8.7.5
Relevant Packages:
  next: 14.0.0
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 4.6.4
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Turbopack (--turbo)

### Additional context

I have installed a template using create-next-app and attempted to use --turbo for compiling the application with slight changes to use multiple root layouts. The exact same setup works without --turbo flag. Same scenario applies to latest canary release.

**Comment by razenization** [comment]
Seems like it got fixed with v14.0.1. Can someone from the team confirm? I believe #57647 addresses that.

**Comment by ForsakenHarmony** [comment]
@razenization yes this should be fixed with #57647
